### PR TITLE
replace tests skip with soft xfail

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

have then skip is not best practice as people look and  see test for a case but easily miss it is skipped so lets let then fail or remove completely...

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18486.org.readthedocs.build/en/18486/

cc @justusschock @awaelchli @carmocca @borda

### CODE REVIEW

File: tests/tests_pytorch/trainer/test_dataloaders.py [hunk 1]
```diff
@@ -341,7 +341,9 @@ def test_dataloaders_with_limit_val_batches(tmpdir, dataset):
     assert epoch_cb.val_batches_seen == limit_val_batches * max_epochs
 
 
-@pytest.mark.skip()
+@pytest.mark.xfail(
+    strict=False,
+)
```
Comment (Borda) [inline tests/tests_pytorch/trainer/test_dataloaders.py:346]:
```suggestion
@pytest.mark.xfail(strict=False)
```

File: tests/tests_pytorch/trainer/test_dataloaders.py [hunk 2]
```diff
@@ -385,7 +387,9 @@ def test_datasets_dataloaders_with_limit_num_batches(tmpdir, dataset):
     assert epoch_cb.test_epoch_count == 1
 
 
-@pytest.mark.skip()
+@pytest.mark.xfail(
+    strict=False,
+)
```
Comment (Borda) [inline tests/tests_pytorch/trainer/test_dataloaders.py:392]:
```suggestion
@pytest.mark.xfail(strict=False)
```

File: tests/tests_pytorch/trainer/test_dataloaders.py [hunk 3]
```diff
@@ -385,7 +385,7 @@ def test_datasets_dataloaders_with_limit_num_batches(tmpdir, dataset):
     assert epoch_cb.test_epoch_count == 1
 
 
-@pytest.mark.skip()
+@pytest.mark.xfail(strict=False)
```
Comment (awaelchli) [inline tests/tests_pytorch/trainer/test_dataloaders.py:388]: here too
```suggestion
```

File: tests/tests_pytorch/trainer/test_dataloaders.py [hunk 4]
```diff
@@ -341,7 +341,7 @@ def test_dataloaders_with_limit_val_batches(tmpdir, dataset):
     assert epoch_cb.val_batches_seen == limit_val_batches * max_epochs
 
 
-@pytest.mark.skip()
+@pytest.mark.xfail(strict=False)
```
Comment (awaelchli) [inline tests/tests_pytorch/trainer/test_dataloaders.py:344]: This can be removed, the test passes

```suggestion
```

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Pull request merged**